tests/foreman/api/test_virt_who_config.py

```diff
@@ -0,0 +1,408 @@
+    @tier1
+    def test_negative_role_reporter(self):
+        """Verify virt-who Reporter role
+
+        :id: 7dee0965-9ec4-4d76-a6ae-f2eec1960bac
+
+        :steps:
+            1. Verify the user can do no other actions other then those
+               in test_positive_role_reporter
+
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_role_viewer(self):
+        """ Verify virt-who viewer role
+
+        :id: 43314ae3-6768-44b5-a6bb-ee64ae381cd0
+
+        :steps:
+            1. Create a user with ONLY the virt-who Viewer role.
+            2. Verify the user can view virt-who configurations.
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_role_viewer(self):
+        """ Verify virt-who viewer role
+
+        :id: 6919a748-423e-4843-bafe-eb98b7159c90
+
+        :steps:
+            1. Create a user with ONLY the virt-who Viewer role.
+            2. Verify the user CANNOT delete or modify virt-who configurations
+            3. Verify the user can do no other actions other then those
+               in test_positive_role_viewer
+
+        """
+        pass
+
+
+class VirtWhoConfigUpgrade(APITestCase):
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_satellite_upgrade(self):
+        """ Satellite upgrade
+
+        :id: 0e301c08-8bef-4bea-a690-d4b0760949e8
+
+        :steps:
+            1. Start with a satellite version before the virt-who config plugin
+            2. Configure virt-who server with virtualization provider (VP1) to
+               send reports to the satellite.
+            3. Upgrade the satellite to a version with virt-who config plugin.
+            4. Configure a new virtualization provider (VP2) with a VDC
+               subscription
+            5. Using the virt-who plugin, create a configuration for the
+               new hypervisor.
+            6. Deploy the configuration to the virt-who server
+            7. Create a guest on VP1
+            8. Create a guest on VP2
+            9. verify that that reports on VP1 and VP2 are correct.
+
+        """
+
+
+class VirtWhoConfigGeneralTestcase(TestCase):
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_hypervisors(self):
+        """End to End scenarios. For all supported Hypervisors
+           (Libvirt, vmware, RHEV, Hyper-V, Xen)
+
+        :id: ce67645a-8435-4e2a-9fd5-c54f2a11c44b
+
+        :steps:
+            1. Associate a VDC subscription to the hypervisor host
+            2. Create a Virt-who configuration for the hypervisor.
+            3. Deploy the virt-who configuration
+            4. Create VM1 on the hypervisor
+            5. Create activation key with Content view, but DO NOT
+               associate subscription.
+            6. Register  VM using the activation key.
+            7. Wait until the next report comes to the satellite
+            8. Verify the VM is report to the Satellite, and the VDC
```

Comment:
this should go to :expectedresults: section. Make sure that every test case should have the expected results.